packages/graphql/src/translate/utils/resolveTree.ts

```diff
@@ -0,0 +1,68 @@
+/*
```

Comment:
Great idea on adding a set of utilities for handling the resolveTree :clap:

Reply:
I think some things can be unnecessarily complicated since most things are with respect to a "`field`" which has two meanings here: a field of the node and `Record<string, ResolveTree>`. Simply manipulating a set of `ResolveTree`s will make this easier.

In a recent refactor (https://github.com/neo4j/graphql/commit/bd391a851360fa7bcda758abc02cd90ab172a18c) `createProjectionAndParams` now takes a `ResolveTree`. It would make things easier to reason about if the reducer in that function were to also be refactored to take `ResolveTree[]` as opposed to a "field" of `Record<string, ResolveTree>`. This would be even more trivial by only requiring a change from `key` to `field.alias` which is what `key` is anyway by construction.

Thoughts on this?

Reply:
Refactor: https://github.com/dmoree/graphql/compare/feature/ignore-directive-required-fields...dmoree:refactor/field-to-resolve-tree

Note: I also changed the names of the utility functions and added unit tests.

Reply:
Hi @dmoree Your changes in that refactor make a lot of sense, I think overall it is an improvement :+1: 
My only minor, (non-blocking) comment on those would be to use a variable for `field.alias` in the reducer function just to avoid accessing to `.alias` all the time.
Feel free to add that commit to this PR so I can make a last review on those changes and hopefully merge this soon

Reply:
Refactored in https://github.com/neo4j/graphql/pull/842/commits/339cbc2928aee6e30ccd6b5794355d0c389c4b31